types/react-relay/relay-hooks/helpers.d.ts

```diff
@@ -5,10 +5,15 @@ A TS file to help with the construction of the official Relay (flow) types.
 import { FragmentReference } from 'relay-runtime';
 import { EntryPoint } from './EntryPointTypes';
 
-export type KeyType<TData = unknown> = Readonly<{
-    ' $data'?: TData | undefined;
-    ' $fragmentRefs': FragmentReference;
-}>;
+export type KeyType<TData = unknown> = Readonly<
+    | {
+        ' $data'?: TData | undefined;
+        ' $fragmentRefs': FragmentReference;
+    }
+    | {
+        ' $data'?: TData | undefined;
+        ' $fragmentSpreads': FragmentReference;
+    }>;
```

Comment:
@sibelius @AndrewIngram

Not too familiar with this new pusudo property — will this work? As in is it a union, or can it be both?

Reply:
Confirmed that this does work with Relay 13, not confirmed that it still works with relay 12 though

Reply:
Doe DefinitelyTyped have a way to have different types per relay-runtime version? We could make this change just for Relay 13+ ?

Reply:
Nope, the only way is to "fork" by duplicating the folders. Which is more than okay. 

But maybe @orta has a few tricks

Reply:
It can't do this with beta/rcs (or at least I don't have the time to go add support for that, it's usually done for keeping older types around in projects rather than the other way around) - I've got this tested in a relay 12 app, so I think it's good to go. 

My apps are pretty trivial so maybe I've missed something but it's trivial to revert if so

Reply:
That's fair. We've got some tests here that also pass. So I'm personally not offended by merging this. 

But whilst we've got relay folk in the thread, at the top of your head. Anything else we should go explore in making types line up to 13?